File: runtime/CSharp/src/LexerEmptyModeStackException.cs
namespace Antlr4.Runtime
{
    public class LexerEmptyModeStackException : LexerException
    {
        public LexerEmptyModeStackException(Lexer lexer, ICharStream input, int startIndex, int length)
            : base(lexer, input, startIndex, length)
        {
        }

        public override string GetErrorMessage(string input)
        {
            return "Unable to pop mode because mode stack is empty at: '" + input + "'";
        }
    }
}

File: runtime/Java/src/org/antlr/v4/runtime/LexerEmptyModeStackException.java
package org.antlr.v4.runtime;

public class LexerEmptyModeStackException extends LexerException {
	public LexerEmptyModeStackException(Lexer lexer, IntStream input, int startIndex, int length) {
		super(lexer, input, startIndex, length);
	}

	@Override
	public String getErrorMessage(String input) {
		return "Unable to pop mode because mode stack is empty at: '" + input + "'";
	}
}
